src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -144,59 +148,57 @@ std::string GetDomainFromHostName(const char * hostnameWithDomain)
-        if (pos != std::string::npos)
-        {
-            // Return the domain name between the last 2 occurences of '.' including the trailing dot'.'.
-            return std::string(hostname.substr(pos + 1, last_pos));
-        }
-    }
-    return std::string();
-}
+    // Find the first occurence of '.'
+    size_t first_pos = hostname.find(".");
 
-Global<MdnsContexts> MdnsContexts::sInstance;
+    // if not found, return empty string
+    VerifyOrReturnValue(first_pos != std::string::npos, std::string());
 
-namespace {
+    // Get a substring after the first occurence of '.' to the end of the string
+    return hostname.substr(first_pos + 1, hostname.size());
+}
 
 /**
- * @brief Callback that is called when the timeout for resolving on the kOpenThreadDot domain has expired.
+ * @brief Callback that is called when the timeout for resolving on the kSrpDot domain has expired.
  *
  * @param[in] systemLayer The system layer.
  * @param[in] callbackContext The context passed to the timer callback.
  */
-void OpenThreadTimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
+void SrpTimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
 {
-    ChipLogProgress(Discovery, "Mdns: Timer expired for resolve to complete on the open thread domain.");
+    ChipLogProgress(Discovery, "Mdns: Timer expired for resolve to complete on the srp domain.");
     auto sdCtx = static_cast<ResolveContext *>(callbackContext);
     VerifyOrDie(sdCtx != nullptr);
-
-    if (sdCtx->hasOpenThreadTimerStarted)
-    {
-        sdCtx->Finalize();
-    }
+    sdCtx->Finalize();
 }
 
 /**
- * @brief Starts a timer to wait for the resolution on the kOpenThreadDot domain to happen.
+ * @brief Starts a timer to wait for the resolution on the kSrpDot domain to happen.
  *
  * @param[in] timeoutSeconds The timeout in seconds.
  * @param[in] ResolveContext The resolve context.
  */
-void StartOpenThreadTimer(uint16_t timeoutInMSecs, ResolveContext * ctx)
+CHIP_ERROR StartSrpTimer(uint16_t timeoutInMSecs, ResolveContext * ctx)
 {
-    VerifyOrReturn(ctx != nullptr, ChipLogError(Discovery, "Can't schedule open thread timer since context is null"));
-    DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds16(timeoutInMSecs), OpenThreadTimerExpiredCallback,
-                                          reinterpret_cast<void *>(ctx));
+    VerifyOrReturnValue(ctx != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    return DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds16(timeoutInMSecs), SrpTimerExpiredCallback,
+                                                 reinterpret_cast<void *>(ctx));
 }
 
+/**
+ * @brief Cancels the timer that was started to wait for the resolution on the kSrpDot domain to happen.
+ *
+ * @param[in] ResolveContext The resolve context.
+ */
+void CancelSrpTimer(ResolveContext * ctx)
+{
+    DeviceLayer::SystemLayer().CancelTimer(SrpTimerExpiredCallback, reinterpret_cast<void *>(ctx));
```

Comment:
Why do you need the cast here?  Just passing `ctx` as the second arg should work.  Same in StartTimer....